src/views/UserEdit/UserEdit.js

```diff
@@ -185,7 +186,7 @@ class UserEdit extends React.Component {
           return mutator.perms.POST({ userId: user.id, permissions: [] });
         })
         .then(() => {
```

Comment:
I suggest to simplify logic and have just one mutator call with a check for credentials in the first then.

```suggestion
  create = ({ requestPreferences, creds, ...userFormData }) => {
    const { mutator, history, location: { search } } = this.props;
    const userData = cloneDeep(userFormData);
    const credentialsAreSet = userData.username;
    const user = { ...userData, id: uuid() };
    user.personal.addresses = toUserAddresses(user.personal.addresses);

    mutator.records.POST(user)
      .then((response) => {
        if (!credentialsAreSet) {
          return response;
        }

        const credentials = {
          password: '',
          ...creds,
          username: userData.username,
        };

        return mutator.creds.POST(Object.assign(credentials, { userId: user.id }));
      })
      .then(() => {
        this.createRequestPreferences(requestPreferences, user.id);
        return mutator.perms.POST({ userId: user.id, permissions: [] });
      })
      .then(() => {
        history.push(`/users/preview/${user.id}${search}`);
      });
    }
  }
```